tests/unit/test_cli.py

```diff
@@ -0,0 +1,263 @@
+    mock_acquire_lock.return_value = fd
+    mock_os.path.isdir.return_value = False
+    run_updates(False, True)
+    mock_release_lock.assert_called_once_with(fd)
+
+
+@patch("ublue_update.cli.os")
+@patch("ublue_update.cli.get_active_sessions")
+@patch("ublue_update.cli.acquire_lock")
+@patch("ublue_update.cli.transaction_wait")
+@patch("ublue_update.cli.subprocess.run")
+@patch("ublue_update.cli.log")
+@patch("ublue_update.cli.pending_deployment_check")
+@patch("ublue_update.cli.cfg")
+@patch("ublue_update.cli.release_lock")
+@patch("ublue_update.cli.notify")
+def test_run_updates_system(
+    mock_notify,
+    mock_release_lock,
+    mock_cfg,
+    mock_pending_deployment_check,
+    mock_log,
+    mock_run,
+    mock_transaction_wait,
+    mock_acquire_lock,
+    mock_get_active_sesions,
+    mock_os,
+):
+    mock_os.getuid.return_value = 0
+    mock_acquire_lock.return_value = 3
+    output = MagicMock(stdout=b"test log")
+    output.returncode = 1
+    mock_run.return_value = output
+    mock_pending_deployment_check.return_value = True
+    mock_cfg.dbus_notify.return_value = True
+    run_updates(True, True)
+    mock_notify.assert_any_call(
+        "System Updater",
+        "System passed checks, updating ...",
+    )
+    mock_run.assert_any_call(
+        [
+            "/usr/bin/topgrade",
+            "--config",
+            "/usr/share/ublue-update/topgrade-system.toml",
+        ],
+        capture_output=True,
+    )
+    mock_notify.assert_any_call(
+        "System Updater",
+        "System update complete, pending changes will take effect after reboot. Reboot now?",
+        ["universal-blue-update-reboot=Reboot Now"],
+    )
+
+
+@patch("ublue_update.cli.os")
+@patch("ublue_update.cli.get_active_sessions")
+@patch("ublue_update.cli.acquire_lock")
+@patch("ublue_update.cli.transaction_wait")
+@patch("ublue_update.cli.subprocess.run")
+@patch("ublue_update.cli.log")
+@patch("ublue_update.cli.pending_deployment_check")
+@patch("ublue_update.cli.cfg")
+@patch("ublue_update.cli.release_lock")
+@patch("ublue_update.cli.notify")
+def test_run_updates_system_reboot(
+    mock_notify,
+    mock_release_lock,
+    mock_cfg,
+    mock_pending_deployment_check,
+    mock_log,
+    mock_run,
+    mock_transaction_wait,
+    mock_acquire_lock,
+    mock_get_active_sesions,
+    mock_os,
+):
+    mock_os.getuid.return_value = 0
+    mock_acquire_lock.return_value = 3
+    output = MagicMock(stdout=b"test log")
+    output.returncode = 1
+    mock_run.return_value = output
+    mock_pending_deployment_check.return_value = True
+    mock_cfg.dbus_notify.return_value = True
+    reboot = MagicMock(stdout=b"universal-blue-update-reboot")
+    mock_notify.side_effect = [None, reboot]
+    run_updates(True, True)
+    mock_notify.assert_any_call(
+        "System Updater",
+        "System passed checks, updating ...",
+    )
+    mock_run.assert_any_call(
+        [
+            "/usr/bin/topgrade",
```

Comment:
to alleviate the problem with these args needing to be test file maybe it would be better to store them as constants as to consolodate the changes needed if we move forward with testing the args

Reply:
I agree with the sentiment, but the practicality of it is pretty low imo.  many of those mocked runs are actually different from each other so making them constants only helps a little and reduces the readability.

Reply:
what i will say is that i much prefer if we go the constant route to host the constants in the src itself and import them with the functions